### PR TITLE
Support expose status for any sequencer(backend)

### LINKED ISSUE #594: AnySequencer: exposing "status" instead of failure
/kind feature

**Description:**
As tested by @sswarna, when no conditions of its dependencies are met, AnySequencer simply fails. That is not the expected behaviour (or at least not the default one) --- it is expected that, like conditions, AnySequencer will be marked as "skipped".

Because Tekton currently provides no possibility to "skip" a task or even a custom task after it started execution, as "skipping" isn't a status state, it's a concept of "task that LACKS a task run" or "custom task that LACKS a run" --- it cannot be handled directly.

Due to a recent feature in Tracking API, any component exposing status output and Tekton result may modify its "Completed" Tekton status to some other value. It has been checked that it works with "Skipped" too. Other tasks can react to it via when clauses, treating a task with "Skipped" status as not-completed and therefore the skip will be propagated as expected.

That requires the following change to the AnySequencer:

- name: any-sequencer-0001
  taskSpec:
    steps:
    - name: main
      image: dspipelines/any-sequencer:latest
      command:
      - any-taskrun
      args:
      - --namespace
      - $(params.pipelineRun-namespace)
      - --prName
      - $(params.pipelineRun-name)
      - --statusPath
      - $(results.status.path)
      - --taskList
      - condition-cel-2,condition-cel-3,condition-cel
      - -c
      - results_condition-cel-2_outcome == 'true'
      - -c
      - results_condition-cel-3_outcome == 'true'
      - -c
      - results_condition-cel_outcome == 'true'
      env:
        (...)
    metadata:
      (...)
    params:
    - name: pipelineRun-name
    - name: pipelineRun-namespace
  params:
  - name: pipelineRun-name
    value: $(context.pipelineRun.name)
  - name: pipelineRun-namespace
    value: $(context.pipelineRun.namespace)
  results:
  - name: status
    description: /tmp/outputs/status/data
Note the additional result status and the additional flag --statusPath. An additional statusMessage field should be provided too, to explain the reason for the generated status.

As noted in the comment in the original issue, failure on no-dependency-condition-matching may be an optional feature. For that, I imagine two additional optional flags:

skipping_policy --- determines for the AnySequencer reacts to no-dependency-condition-matching case. "skip_on_no_match" or "error_on_no_match"
error_policy --- the standard field, either "fail_on_error" or "continue_on_error". On "continue_on_error", a status with value "Failed" may be generated for skipping_policy set to "error_on_no_match". For "fail_on_error" and skipping_policy set to "error_on_no_match", the AnySequencer should truly fail in the Tekton terms, as it does now.

### PR DESCRIPTION
**Pull request by pugangxa**
**Which issue is resolved by this Pull Request:** 
Resolves #594 

The backend changes, will do more test.

### CODE REVIEW

**Comment by pugangxa** [comment]
@Tomcli @fenglixa Please help take a look, thanks.

File: sdk/FEATURES.md [hunk 1]
```diff
@@ -142,7 +142,13 @@ To see how the Python SDK provides this feature, refer to the examples below:
...
+When any one of the task dependencies completes successfully and the conditions meet, the dependent task will be started. Order of execution of the dependencies doesn’t matter, and the pipeline doesn't wait for all the task dependencies to complete before moving to the next step. Condition can be applied to enforce the task dependencies completes as expected. The condition expression should be the same format as is in Kubeflow ConditionOperator, and the result of containerOps can be used in expression. Notice the expression should only contain results from only one task because the purpose here is to check the simple condition for the task's output when a task complete. And also the operand in the expression should be int or string, other python types will be transferred to string automatically. 
+
+The exit status of any-sequencer is configurable through parameters, `statusPath` must be sepecified for any-sequencer to write status to, then use `skippingPolicy` and `errorPolicy`.
```
Comment (Tomcli) [inline sdk/FEATURES.md:147]:
```suggestion
The exit status of any-sequencer is configurable through parameters, `statusPath` must be specified for any-sequencer to write status to, then use `skippingPolicy` and `errorPolicy`.
```

**Review by Tomcli — COMMENTED** [review]
thanks @pugangxa, it works within our CI and I will test the new feature using the new changes in #598

**Comment by Tomcli** [comment]
tested with #598 and looks good to me. Just need to fix the typo in the readme ^^

**Comment by jinchihe** [comment]
Overall, looks good to me, thanks @pugangxa

File: sdk/FEATURES.md (hunk 1 above)
Comment (pugangxa) [reply sdk/FEATURES.md:147]: Thanks Tommy, typo fixed.

**Comment by fenglixa** [comment]
/lgtm
Thanks Gang

**Pull request merged**